### ISSUE TITLE
about process steps

### ISSUE THREAD

**Issue by LucyFann** [wxl1999/UniCRS#1]
Hey Xiaolei. First of all, thanks for your work! I have successfully run your code on Github, but I have a few questions about the preprocessing code. 
1. I got very high recall@k scores in pretrained-prompt model.
  {'test/recall@1': 0.5659283956497578, 'test/recall@10': 0.9083115027387473, 'test/recall@50': 0.9323648487735176, 'test/ndcg@1': 0.5659283956497578, 'test/ndcg@10': 0.757117269984917, 'test/ndcg@50': 0.7626062380924029, 'test/mrr@1': 0.5659283956497578, 'test/mrr@10': 0.7063158638961657, 'test/mrr@50': 0.7075750767791935, 'test/loss': 2.209772330341917, 'epoch': 4}
Is this due to the file preprocessed(test_data_processed.jsonl) through process.py containing user and system responses? So this recall@k is not accurate? 

2. I wonder if it is correct to understand the code for the three preprocessed datasets in this way.
(1) the 'process.py' is for extracting user and system response and their context for semantic fusion.
(2) the 'process_mask.py' is for extracting system response for conv prompt.
(3) the 'merge.py' is for merge the template by conv prompt model and items for rec prompt.

It would be very grateful indeed if you can give me anyhelp.
Best regards.
siqi

**Comment by wxl1999** [comment]
Hi, LucyFann. Thanks for your attention on my work!

1. This is because each input contains the target text mentioning the entity which we require the model to predict. By doing so, the model can associate the entity in kg with that in text (PLM). `Recall@k` should be high so that the model are well pre-trained.
2. (1)(2) are true; for (3), `merge.py` is used to just merge the template, but extract items for rec prompt training, you may not state so accurately.

Hope these will help you!

**Comment by LucyFann** [comment]
Thank you for the clarification.

**Comment by LucyFann** [comment]
Hey, xiaolei.  I found that you also calculate bleu@k in your evaluator, but why do most CRS baselines (UniCRS, KGSF, C2CRS...) use DIST@k as the metric for CRS?

**Comment by wxl1999** [comment]
Hi, LucyFann. About this question, you can see the ` Evaluation Metrics` section of KGSF paper for details.

**Comment by LucyFann** [comment]
ok! Thanks!
